docs/source/usage/plugins/shadowgraphy.rst

```diff
@@ -0,0 +1,108 @@
+========================================= ==============================================================================================================================
+``--shadowgraphy.start``                  Step to start the plugin and with this the time integration.
+``--shadowgraphy.duration``               Duration of shadowgraphy calculation in simulation time steps.
+                                          The plugin will be called ``duration / params::tRes`` times.
+                                          The final plugin call (and the file output) happens at the time defined in ``.period`` plus the duration.
+``--shadowgraphy.slicePoint``             Specifies at what ratio of the total depth of the z dimension, the slice for the field extraction should be set.
+                                          The value given should lie between ``0.0`` and ``1.0``.                            
+``--shadowgraphy.focusPos``               Focus position of lens system relative to slicePoint. The focus position is given in SI units. 
+``--shadowgraphy.fileName``               Output file prefix for openPMD output.          
+``--shadowgraphy.ext``                    Backend for openPMD output.
+``--shadowgraphy.fourierOutput``          If enabled, the fields will also be stored on disk in in ``(x, y, \omega)`` Fourier space in an openPMD file.
+========================================= ==============================================================================================================================
+
+.. note::
+   Currently the plugin only supports an integration slice in the xy-plane, which means that for probing setups the probe pulse should propagate in z direction.
+   The moving window can't be activated or deactivated during the plugin integration loop.
+
+
+Output
+^^^^^^
+Plot the first shadowgram that is stored in the simulation output directory ``simOutput``.
+
+.. code:: python
+
+   import matplotlib.pyplot as plt
+   import numpy as np
+   import openpmd_api as io
+
+
+   def load_shadowgram(series):
+      i = series.iterations[[i for i in series.iterations][0]]
+
+      shadowgram_tmp = i.meshes["shadowgram"][io.Mesh_Record_Component.SCALAR].load_chunk()
+      unit = i.meshes["shadowgram"].get_attribute("unitSI")
+      series.flush()
+
+      return shadowgram_tmp * unit
+
+
+   def load_meshgrids(series):
+      i = series.iterations[[i for i in series.iterations][0]]
+
+      xspace_tmp = i.meshes["Spatial positions"]["x"].load_chunk()
+      xunit = i.meshes["Spatial positions"]["x"].get_attribute("unitSI")
+      series.flush()
+      xspace = xspace_tmp * xunit
+
+      yspace_tmp = i.meshes["Spatial positions"]["y"].load_chunk()
+      yunit = i.meshes["Spatial positions"]["y"].get_attribute("unitSI")
+      series.flush()
+      yspace = yspace_tmp * yunit
+
+      return np.meshgrid(xspace, yspace)
+
+
+   path = "/home/carste06/SCRATCH/runs/2024_ShadowgraphyDev/movingwindow11/simOutput"
```

Comment:
Please use a generic path here

Reply:
Oh, this came from testing the code :D Done.